### PR TITLE
Adding Text pattern to ToolStripTextBox

### LINKED ISSUE #3860: [Accessibility] ToolStripTextBox must support Text pattern
- Original Bug: [1155254](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1155254)
* .NET Core Version: 5.0.0-rc.1.20431.5

* Have you experienced this same bug with .NET Framework?: Yes

**Problem description:**
- ToolStripTextBox doesn't support Text pattern
- A ScreenReader can't announce separate navigated, selected, or added characters in ToolStripTextBox 

**Expected behavior:**
- ToolStripTextBox must support Text pattern (may be checked using Inspect)
- A ScreenReader announces separate navigated, selected, or added characters in ToolStripTextBox 

**Minimal repro:**
- Create a .Net 5.0 app
- Add a ToolStrip with a ToolStripTextBox item
- Run the app and check ToolStripTextBox using Inspect or Narrator

### PR DESCRIPTION
**Pull request by vladimir-krestov**
Fixes #3860 (BindingNavigator too)
Orginal Bug: [1155254](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1155254)
Based on PR #2701

## Proposed changes

- Enable **Text pattern** support for `ToolStripTextBoxControlAccessibleObject`

## Customer Impact

- Visually impaired users will be able to read and interact with the text content of a ToolStripTextBox

## Regression? 

- No

## Risk

- Minimal

### Before

- Narrator does not announce the text navigation and selection of a ToolStripTextBox

### After

- Narrator announces the text content of a ToolStripTextBoxs (navigation, selection)

## Test methodology 

- Manual testing
- CTI
- Unit tests

## Accessibility testing  
- Using Narrator and Inspect
   
     

## Test environment(s) 

- .Net 5.0 Version: 5.0.0-rc.1.20431.5
- Microsoft Windows [Version 10.0.19041.450]

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3865)

### CODE REVIEW

**Comment by vladimir-krestov** [comment]
Testers approved the fix ✔️ 
There are 2 known issues that will be fixed as a separate fix for DataGridViewTextBoxCell and PropertyGridView Edit too: 
- [#2794](https://github.com/dotnet/winforms/issues/2794): Narrator doesn't announce a text mouse selection (a keyboard selection works well) 
- Narrator doesn't show the blue rectangle around an added character when typing

But the main Issue #3860 was fixed

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripTextBox.ToolStripTextBoxControlAccessibleObject.cs [hunk 1]
```diff
@@ -10,32 +11,37 @@ public partial class ToolStripTextBox
     {
         private class ToolStripTextBoxControlAccessibleObject : ToolStripHostedControlAccessibleObject
         {
+            private readonly TextBoxBase _owningTextBoxBase;
+            private readonly TextBoxBaseUiaTextProvider _textProvider;
+
             public ToolStripTextBoxControlAccessibleObject(TextBox toolStripHostedControl, ToolStripControlHost? toolStripControlHost) : base(toolStripHostedControl, toolStripControlHost)
             {
+                _owningTextBoxBase = toolStripHostedControl;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripTextBox.ToolStripTextBoxControlAccessibleObject.cs:19]: Test for `toolStripHostedControl is null`?

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripTextBox.ToolStripTextBoxControlAccessibleObject.cs [hunk 2]
```diff
@@ -10,11 +10,11 @@ public partial class ToolStripTextBox
     {
         private class ToolStripTextBoxControlAccessibleObject : ToolStripHostedControlAccessibleObject
         {
-            public ToolStripTextBoxControlAccessibleObject(Control toolStripHostedControl, ToolStripControlHost toolStripControlHost) : base(toolStripHostedControl, toolStripControlHost)
+            public ToolStripTextBoxControlAccessibleObject(TextBox toolStripHostedControl, ToolStripControlHost? toolStripControlHost) : base(toolStripHostedControl, toolStripControlHost)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripTextBox.ToolStripTextBoxControlAccessibleObject.cs:17]: ❗ `ToolStripHostedControlAccessibleObject` must be NRT annotated before any of its descendants (i.e. `ToolStripTextBoxControlAccessibleObject`) can be annotated.

**Review by RussKie — CHANGES_REQUESTED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripTextBox.ToolStripTextBoxControlAccessibleObject.cs (hunk 2 above)
Comment (vladimir-krestov) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripTextBox.ToolStripTextBoxControlAccessibleObject.cs:17]: Annotated NRT for `ToolStripHostedControlAccessibleObject`. 
I checked - related types: `ToolStripNumericUpDownAccessibleObject` and `ToolStripTextBoxControlAccessibleObject`

**Comment by vladimir-krestov** [comment]
@JeremyKuhne, I fixed your #3853 review points here for .Net 5.0. Could you please check if I made this correctly?

**Comment by JeremyKuhne** [comment]
Handling of handles looks much better.

**Review by RussKie — APPROVED** [review]

**Review by RussKie — APPROVED** [review]

**Pull request merged**